docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+
+```jldoctest dataframe
+julia> df1[!, :Customers] .= ["Rohit", "Akshat", "Rahul", "Aayush", "Prateek", "Anam"] # allocates a new column `:Customers` and adds it
+6-element Vector{String}:
+ "Rohit"
+ "Akshat"
+ "Rahul"
+ "Aayush"
+ "Prateek"
+ "Anam"
+
+julia> df1[:, :City] .= ["Kanpur", "Lucknow", "Bhuvneshwar", "Jaipur", "Ranchi", "Dehradoon"] # allocates the column because `:City` is not present in `data`
+6-element Vector{String}:
+ "Kanpur"
+ "Lucknow"
+ "Bhuvneshwar"
+ "Jaipur"
+ "Ranchi"
+ "Dehradoon"
+
+julia> df1
+6×5 DataFrame
+ Row │ Age    Sex          Job    Customers  City
+     │ Int64  String       Int64  String     String
+─────┼───────────────────────────────────────────────────
+   1 │    85  male             4  Rohit      Kanpur
+   2 │    89  female           5  Akshat     Lucknow
+   3 │    78  male             7  Rahul      Bhuvneshwar
+   4 │    58  transgender      8  Aayush     Jaipur
+   5 │    96  female           2  Prateek    Ranchi
+   6 │    68  female           1  Anam       Dehradoon
+```
+
+Assignment of a scalar to a data frame can be done in ranges using broadcasting:
+
+```jldoctest dataframe
+julia> df1[:, 3] .= 4 # an in-place replacement of values stored in column number 3 by 4
+6-element view(::Vector{Int64}, :) with eltype Int64:
+ 4
+ 4
+ 4
+ 4
+ 4
+ 4
+
+julia> df1
+6×5 DataFrame
+ Row │ Age    Sex          Job    Customers  City
+     │ Int64  String       Int64  String     String
+─────┼───────────────────────────────────────────────────
+   1 │    85  male             4  Rohit      Kanpur
+   2 │    89  female           4  Akshat     Lucknow
+   3 │    78  male             4  Rahul      Bhuvneshwar
+   4 │    58  transgender      4  Aayush     Jaipur
+   5 │    96  female           4  Prateek    Ranchi
+   6 │    68  female           4  Anam       Dehradoon
+```
+
+Here, `!` get us columns without copying and when setting columns it replaces them while 
+`:` get us columns with copying and when setting columns it does this in-place. 
+
+```jldoctest dataframe
+julia> df1[:, :Age] .= "Economics"
+ERROR: MethodError: Cannot `convert` an object of type String to an object of type Int64
+
+julia> df1[!, :Age] .= "Economics"
+6-element Vector{String}:
+ "Economics"
+ "Economics"
+ "Economics"
+ "Economics"
+ "Economics"
+ "Economics"
+
+julia> df1
+6×5 DataFrame
+ Row │ Age        Sex          Job    Customers  City
+     │ String     String       Int64  String     String
+─────┼───────────────────────────────────────────────────────
+   1 │ Economics  male             4  Rohit      Kanpur
+   2 │ Economics  female           4  Akshat     Lucknow
+   3 │ Economics  male             4  Rahul      Bhuvneshwar
+   4 │ Economics  transgender      4  Aayush     Jaipur
+   5 │ Economics  female           4  Prateek    Ranchi
+   6 │ Economics  female           4  Anam       Dehradoon
+```
+
+In most cases above, as you can see, for getting a column or assigning to a column instead of `df1[!, :col]` 
```

Comment:
where can it be seen in these examples?

Reply:
and why do you write about it in a section about broadcasting?

Reply:
I had replaced a section then I had missed it to remove.